### DISCUSSION TITLE
Why doesn't strictExportPresence default to true?

### DISCUSSION THREAD

**Discussion by sppatel** [webpack/webpack#14659] [Q&A] [answered]
We were burned by an issue where the an unresolved reference was present and the warning that is currently shown was only seen in our CI logs.  But given that it was a WARNING and not an ERROR, the build passed and led to a production build w/ a unresolved reference to a file.

So the question is why this property doesn't just default to true for the primary case? Is it more of a common scenario where you'd want this false, if so what are those scenarios?

**Comment by sokra** [comment, marked as answer]
When webpack 5 was released, typescript had a lot issues with invalid exports (caused by type reexports in isolated mode), so we had to made it a warning. Now we can't change it without a major release...

I'll change the default for webpack 6 (currently under the `experiments.futureDefaults`)

**Reply by sokra** [reply]
btw. it already defaults to strict when using `"type": "module"` in package.json or `.mjs`

**Reply by sppatel** [reply]
Appreciate the quick response! ❤️

**Reply by sokra** [reply]
https://github.com/webpack/webpack/pull/14660
